modal/experimental/ipython.py

```diff
@@ -0,0 +1,80 @@
+# Copyright Modal Labs 2025
+
+"""This module provides Jupyter/IPython extensions for Modal.
+
+Use in a notebook with `%load_ext modal.experimental.ipython`.
+"""
+
+from IPython.core.magic import Magics, line_magic, magics_class
+
+from ..cls import Cls
+from ..exception import NotFoundError
+from ..functions import Function
+
+
+@magics_class
+class ModalMagics(Magics):
+    @line_magic
+    def modal(self, line):
+        """Lookup a deployed Modal Function or Class.
+
+        **Example:**
+
+        ```python notest
+        %modal from main/my-app import my_function, MyClass as Foo
```

Comment:
Curious if you think there's any risks for overloading the "import" concept here. I.e. in normal Python code, doing

```python
from my_modal_module import MyClass
```

does not give you a hydrated version of that class, just a pointer to the local object that's probably registered to a (non-running) App.

Reply:
> The syntax is "magic" so I'd believe that people probably wouldn't extrapolate to normal behavior from this one

Not all users will know that `%modal` makes the line magic right? Especially if we don't require you to always include `{env}/` it feels possible to mistake that for a Python import statement even if it's doing something different.

Reply:
Hmm I think it's pretty clear that `%modal` makes the line magic. It's an unusual syntax for Python for people who are unfamiliar with notebooks, and it's familiar to people who are familiar with notebooks.